src/Tasks/FileIO/WriteLinesToFile.cs

```diff
@@ -87,6 +88,7 @@ public override bool Execute()
                     {
                         Directory.CreateDirectory(directoryPath);
                         string contentsAsString = buffer.ToString();
+                        MSBuildEventSource.Log.WriteLinesToFileUpToDateStart();
```

Comment:
It makes me uncomfortable to emit this in (the standard) cases where `WriteOnlyWhenDifferent` is unset/false. Should we condition it?

Reply:
It's implicitly conditioned on events being enabled, so it won't be firing tons of traces into the void. Is it just that you want to be clear whether our check is super fast versus only checking WriteOnlyWhenDifferent?

Reply:
Yes, I'd like to be able to distinguish when we even tried to do the up-to-date check.